pkg/logql/engine.go

```diff
@@ -381,9 +385,36 @@ func (q *query) evalSample(ctx context.Context, expr syntax.SampleExpr) (promql_
 	return nil, errors.New("unexpected empty result")
 }
 
-func (q *query) JoinSampleVector(next bool, r StepResult, stepEvaluator StepEvaluator, maxSeries int) (promql_parser.Value, error) {
+func vectorsToSeries(vec promql.Vector) ([]promql.Series, map[uint64]*promql.Series) {
```

Comment:
Do we want a test function for this?

Reply:
this is tested via `TestMappingEquivalence_Instant`, we could add a test specifically for this but it didn't feel necessary